swupdate: update bootloader settings after applying an update

chombourger opened the pull request:
Signed-off-by: Cedric Hombourger <Cedric_Hombourger@mentor.com>

chombourger committed:
swupdate: update bootloader settings after applying an update

Signed-off-by: Cedric Hombourger <Cedric_Hombourger@mentor.com>

chombourger committed:
swupdate: create a single ebg environment per update

A new environment is already created when updating boot settings for the
alternate boot partition. Do not create a new one when suricatta is trying
to set the ustate variable since this would create a new environment for
the current boot partition (which should not be changed!)

Signed-off-by: Cedric Hombourger <Cedric_Hombourger@mentor.com>

Noor-Ahsan reviewed (commented):
@cedric can you add the reason why we need this in commit message for https://github.com/MentorEmbedded/meta-iot2000/pull/14/commits/41389cd70aae5bb70baf06e2be2cfc13b5ee1db4

chombourger commented:
done

chombourger committed:
swupdate: select the platform to be updated on startup

Determine which root file-system partition was booted to determine the
swupdate configuration to be used when updating (either stable,platform0
or stable,platform1)

Signed-off-by: Cedric Hombourger <Cedric_Hombourger@mentor.com>
